### PR TITLE
Add ostream operator template for std::vectors

### PR DESCRIPTION
**Pull request by terhorstd**
Streaming `std::vector`s just streams all elements (given that the elements support `cout << obj;`). This is very useful in many circumstances and should not be implemented each time anew.

### CODE REVIEW

**Comment by jougs** [comment]
@terhorstd: you can either fix the format of the code you're proposing or start another PR to change the formatting guidelines as discussed two weeks ago and to make your code pass the tests. Both options would be fine for me at this point in time.

File: nestkernel/streamers.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		if (not first)
+		{
+			os << ", ";
+		}
+		os << element;
+	}
+	return os << "]";
+}
+
+#endif //ndef STREAMERS_H
```
Comment (heplesser) [inline nestkernel/streamers.h]: Why `ndef`?

File: nestkernel/streamers.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/*
```
Comment (heplesser) [inline nestkernel/streamers.h]: `libnestutil` would be a better location for this file.

File: nestkernel/streamers.h [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ *  streamers.h
```
Comment (heplesser) [inline nestkernel/streamers.h]: I don't like the name `streamers`.  How about `prettyprint`?

File: nestkernel/streamers.h (hunk 2 above)
Comment (jougs) [reply nestkernel/streamers.h]: +1

File: nestkernel/streamers.h (hunk 3 above)
Comment (terhorstd) [reply nestkernel/streamers.h]: The function does not print anything, so I would not call it "*print". The function is an operator with any ostream object as first argument. The term "stream" in this context is C++ convention. ...and just dumping an array to a stream is not "pretty" in my opinion. Would you rather like a different pre-/postfix than "vector[" and "]"?

File: nestkernel/streamers.h (hunk 2 above)
Comment (terhorstd) [reply nestkernel/streamers.h]: I agree

File: nestkernel/streamers.h (hunk 1 above)
Comment (terhorstd) [reply nestkernel/streamers.h]: identifies the expected #if statement. we don't seem to have a consistent convention for `#endif`s. I'll leave this for now. We should set up a convention for this.

File: libnestutil/streamers.h [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+{
+  os << "vector[";
+  bool first = true;
+  for ( const auto& element : vec )
+  {
+    if ( not first )
+    {
+      os << ", ";
+    }
+    os << element;
```
Comment (heplesser) [inline libnestutil/streamers.h]: @terhorstd Don't you need to set `first` to false somewhere?

**Pull request merged**